openedx/features/content_type_gating/field_override.py

```diff
@@ -44,11 +44,21 @@ def get(self, block, name, default):
             if group_access_field is not None:
                 original_group_access = group_access_field.from_json(raw_value)
 
+        group_ids = [settings.CONTENT_TYPE_GATE_GROUP_IDS['full_access']]
         if original_group_access is None:
             original_group_access = {}
+
+            # For Feature Based Enrollments, we want to allow granting access
+            # to all graded problems in a unit at the unit level
+            parent = block.get_parent()
+            if parent.scope_ids.block_type == 'vertical':
+                merged_group_access = parent.merged_group_access
+                if merged_group_access and CONTENT_GATING_PARTITION_ID in merged_group_access:
+                    group_ids = merged_group_access[CONTENT_GATING_PARTITION_ID]
```

Comment:
If `CONTENT_GATING_PARTITION_ID` is in `merged_group_access`, this field override provider should probably just bail out without changing anything (because the rest of the system will make sure the overrides work correctly).